server/channels/app/plugin.go

```diff
@@ -352,18 +352,22 @@ func (ch *Channels) syncPlugins() *model.AppError {
 			}
 			defer bundle.Close()
 
-			var signature filestore.ReadCloseSeeker
 			if *ch.cfgSvc.Config().PluginSettings.RequirePluginSignature {
-				signature, appErr = ch.srv.fileReader(plugin.signaturePath)
+				signature, appErr := ch.srv.fileReader(plugin.signaturePath)
 				if appErr != nil {
 					logger.Error("Failed to open plugin signature from file store.", mlog.Err(appErr))
 					return
 				}
 				defer signature.Close()
+
+				if appErr = ch.verifyPlugin(bundle, signature); appErr != nil {
+					logger.Error("Failed to validate plugin signature", mlog.Err(appErr))
+					return
+				}
```

Comment:
See the discrete PR, but essentially we now check the plugin signature in the caller to `installPluginLocally` to simplify various code paths that don't check it.